sketch/Explorer_ESP_WiFi-localization_v1.2.ino

```diff
@@ -0,0 +1,162 @@
+#include <TheThingsNetwork.h>
+
+#define HEX_CHAR_TO_NIBBLE(c) ((c >= 'a') ? (c - 'a' + 0x0A) : (c - '0'))
+#define HEX_PAIR_TO_BYTE(h, l) ((HEX_CHAR_TO_NIBBLE(h) << 4) + HEX_CHAR_TO_NIBBLE(l))
+
+// Parameters for the esp stream/buffer
+#define SF_ESP8266_DEFAULT_WAIT 5000
+#define SF_ESP8266_BUFFER_SIZE 128
+#define ESP_SERIAL_SPEED 9600
+#define ACCESS_POINTS 3
+#define SF_WIFI_BSSID_SIZE 6
+
+// Set your AppEUI and AppKey
+const char *appEui = "0000000000000000";
+const char *appKey = "00000000000000000000000000000000";
+
+// Replace REPLACE_ME with TTN_FP_EU868 or TTN_FP_US915
+#define freqPlan REPLACE_ME
+
+// Define the used serials
+#define debugSerial SerialUSB
+#define loraSerial Serial2
+#define espSerial Serial
+
+// Prototyping the functions
+char *readLine(uint32_t waitTime = SF_ESP8266_DEFAULT_WAIT);
+bool waitForOK(uint32_t waitTime = SF_ESP8266_DEFAULT_WAIT);
+void sendBssid();
+
+// Declaring variables
+uint8_t payloadLength = ACCESS_POINTS * SF_WIFI_BSSID_SIZE;
+char buffer[SF_ESP8266_BUFFER_SIZE];
+byte aps[ACCESS_POINTS][SF_WIFI_BSSID_SIZE];
+
+// TTN constructor
+TheThingsNetwork ttn(loraSerial, debugSerial, freqPlan);
+
+void setup()
+{
+  // Wait a maximum of 10s for Serial Monitor
+  while (!debugSerial && millis() < 10000)
+    ;
+
+  loraSerial.begin(57600);
+  debugSerial.begin(ESP_SERIAL_SPEED);
+  espSerial.begin(ESP_SERIAL_SPEED);
+
+  debugSerial.println("-- STATUS");
+  ttn.showStatus();
+
+  debugSerial.println("-- JOIN");
+  ttn.join(appEui, appKey);
+}
+
+void loop()
+{
+  uint8_t bssidNumber = 0;
+  uint32_t timeout;
+  // command to esp, set station mode
+  espSerial.println(F("AT+CWMODE=1"));
+  bool x = waitForOK();
+
+  if (!x)
```

Comment:
What is `x`?

- Use clear variable names
- If you don't need the variable, like in this case, just inline `waitForOK()` in the `if`

Reply:
I tried to use `waitForOK()` inline, but somehow it would only return back a `false`. But it did work work when i assigned the value to a variable.